NYPLAudiobookToolkit/Core/LCPSpineElement.swift

```diff
@@ -0,0 +1,124 @@
+    lazy var downloadTask: DownloadTask = {
+        return LCPDownloadTask(spineElement: self)
+    }()
+    
+    lazy var chapter: ChapterLocation = {
+        return ChapterLocation(
+            number: self.chapterNumber,
+            part: 0,
+            duration: self.duration,
+            startOffset: 0,
+            playheadOffset: 0,
+            title: self.title,
+            audiobookID: self.audiobookID
+            )!
+    }()
+    
+    let key: String
+    let chapterNumber: UInt
+    let title: String
+    let url: URL
+    let mediaType: LCPSpineElementMediaType
+    let duration: TimeInterval
+    let audiobookID: String
+    
+    /// Spine element for LCP audiobooks
+    /// - Parameters:
+    ///   - JSON: JSON data for `readingOrder` element in audiobook `manifest.json` file.
+    ///   - index: Element index in `readingOrder`.
+    ///   - audiobookID: Audiobook identifier.
+    init?(JSON: Any?, index: UInt, audiobookID: String) {
+        self.key = "\(audiobookID)-\(index)"
+        self.chapterNumber = index
+        self.audiobookID = audiobookID
+        
+        guard let payload = JSON as? [String: Any],
+            let urlString = payload["href"] as? String,
+            let url = URL(string: urlString)
+            else {
+                ATLog(.error, "LCPSpineElement failed to init from JSON: \n\(JSON ?? "nil")")
+                return nil
+        }
+        
+        self.url = url
+        let defaultTitleFormat = NSLocalizedString("Chapter %@", bundle: Bundle.audiobookToolkit()!, value: "Chapter %@", comment: "Default chapter title")
+        let name = LCPSpineElement.elementName(JSON: payload["title"])
+        self.title = name ?? String(format: defaultTitleFormat, "\(index + 1)")
+        if let duration = payload["duration"] as? Double {
+            self.duration = TimeInterval(duration)
+        } else {
+            self.duration = 0
+        }
+        if let encodingFormat = payload["encodingFormat"] as? String, let mediaType = LCPSpineElementMediaType(rawValue: encodingFormat) {
+            self.mediaType = mediaType
+        } else {
+            self.mediaType = .audioMP3
+        }
+    }
+    
+    /// Title for a spine element from `name` parameter
+    /// - Parameter JSON: `JSON` value for the `name` parameter; can be a string or an array of localized name dictionaries.
+    /// - Returns: String value for `name` paramter or a localized name, if any found; a name matching "en" language code, if found; `nil` otherwise.
+    private static func elementName(JSON: Any?) -> String? {
+        if let stringName = JSON as? String {
+            return stringName
+        } else if let localizedNames = JSON as? [[String: String]] {
+            return localizedName(localizedNames: localizedNames)
+        }
+        return nil
+    }
+    
+    /// Localized name from the array of localizede names based on current application language code.
```

Comment:
typo: `localizede` :)

Reply:
fixed... my butterfly keyboard is doing this sometimes